test/tst_shell.cpp

```diff
@@ -188,6 +188,88 @@ private slots:
 		QVERIFY(onOptionSet.isValid());
 	}
 
+	void CloseEvent_data() {
+		// The status code as carried via msgpack
+		QTest::addColumn<int>("event_status");
+		// The expected process exit status
+		QTest::addColumn<int>("exit_status");
+		// q/cq command
+		QTest::addColumn<QByteArray>("command");
+
+		QTest::newRow("q")
+			<< 0 << 0 << QByteArray("q");
+		QTest::newRow("cq")
+			<< 1 << 1 << QByteArray("cq");
+		QTest::newRow("2cq")
+			<< 2 << 2 << QByteArray("2cq");
+		QTest::newRow("255cq")
+			<< 255 << 255 << QByteArray("255cq");
+		// Overflow except Windows, this generates an exit code 0, but the
+		// original count is passed to us
+		// Commented out as this test doesn't pass on Windows
+//		QTest::newRow("256cq")
+//			<< 256 << 0 << QByteArray("256cq");
+		// a bit of corner case, but nvim exits with
+		// status 0 and provides us count 0 - calling it with nvim -c
+		// actually causes an error with an invalid range, so it can not run
+//		QTest::newRow("-1cq")
+//			<< 0 << QByteArray("-1cq");
```

Comment:
This commented-out segment should be cleaned up before check-in.

Two thoughts:
1) Let's avoid checking in commented-out blocks of code.
2) If this isn't too hard: let's explicitly test the behavior differences between Windows/Linux/MacOS. You can use tst_input* as a pattern to write platform specific code. If this is to much trouble, let's write a clear comment about the behavior differences. Please avoid `#ifdef` ;)

We should have a single clear comment about the differences between Windows vs Linux/Mac.